apps/dashboard/src/components/workflow-editor/steps/delay/delay-amount.tsx

```diff
@@ -0,0 +1,39 @@
+import { NumberInputWithSelect } from '@/components/number-input-with-select';
+
+import { FormLabel } from '@/components/primitives/form/form';
+import { useMemo } from 'react';
+import { TimeUnitEnum } from '@novu/shared';
+import { useStep } from '@/components/workflow-editor/steps/step-provider';
+
+const defaultUnitValues = Object.values(TimeUnitEnum);
+
+const amountKey = 'amount';
+const unitKey = 'unit';
+
+export const DelayAmount = () => {
+  const { step } = useStep();
+  const { dataSchema, uiSchema } = step?.controls ?? {};
+
+  const unitOptions = useMemo(
+    () => (dataSchema?.properties?.[unitKey] as any).enum ?? defaultUnitValues,
+    [dataSchema?.properties]
+  );
+
+  const defaultUnitOption = useMemo(
+    () => (uiSchema?.properties?.[unitKey] as any).placeholder ?? defaultUnitValues[0],
```

Comment:
My personal opinion, because defaultUnitValues[0] is less predictable, we could change the order, and we would get months here.

```suggestion
    () => (uiSchema?.properties?.[unitKey] as any).placeholder ?? defaultUnitValues.SECONDS,
```